src/OpenVocabularyAssociationListBuilder.php

```diff
@@ -113,4 +126,52 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
     $this->messenger->addStatus($this->t('The order of the vocabulary associations has been updated.'));
   }
 
+  /**
+   * Generates an entity types and bundles array the vocabulary associated with.
+   *
+   * @param array $items
+   *   An array of fields associated with.
+   *
+   * @return array
+   *   Array containing the entity type and bundle labels for render.
+   */
+  protected function getAssociatedEntityTypeBundles(array $items): array {
+    $output = [];
+    $bundles_per_entity = [];
+
+    // Process the input array to group bundles by entity type.
+    foreach ($items as $item) {
+      [$entity_type_id, $bundle_id] = explode('.', $item);
+      $bundles_per_entity[$entity_type_id][] = $bundle_id;
+    }
+
+    // Loop through each entity type and get the labels.
+    foreach ($bundles_per_entity as $entity_type_id => $bundle_ids) {
+      // Get the label from the entity type definition.
+      $entity_type = $this->entityTypeManager->getDefinition($entity_type_id);
```

Comment:
[In here](https://github.com/openeuropa/open_vocabularies/blob/master/src/Form/OpenVocabularyAssociationForm.php#L301) we deal with bundle and entity labels with fallbacks, because in all scenarios a label is present. Probably it's better to use a similar approach.

Reply:
I guess you meant, "because not in all scenarios a label is present" .

Reply:
Indeed-o